applications/StructuralMechanicsApplication/custom_elements/bushing_element.h

```diff
@@ -0,0 +1,332 @@
+      */
+    void CalculateMassMatrix(
+        MatrixType& rMassMatrix,
+        const ProcessInfo& rCurrentProcessInfo) override;
+
+    /**
+      * this is called during the assembling process in order
+      * to calculate the elemental damping matrix
+      * @param rDampingMatrix: the elemental damping matrix
+      * @param rCurrentProcessInfo: the current process info instance
+      */
+    void CalculateDampingMatrix(
+        MatrixType& rDampingMatrix,
+        const ProcessInfo& rCurrentProcessInfo) override;
+
+    /**
+     * This function provides the place to perform checks on the completeness of the input.
+     * It is designed to be called only once (or anyway, not often) typically at the beginning
+     * of the calculations, so to verify that nothing is missing from the input
+     * or that no common error is found.
+     * @param rCurrentProcessInfo
+     */
+    int Check(const ProcessInfo& rCurrentProcessInfo) const override;
+
+    void Initialize(const ProcessInfo& rCurrentProcessInfo) override;
+
+    ///@}
+
+protected:
+    ///@name Protected static Member Variables
+    ///@{
+
+    static constexpr unsigned int msNumNodes = 2;
+
+    static constexpr unsigned int msLocalSize = 6;
+
+    static constexpr unsigned int msElementSize = msLocalSize * msNumNodes;
+
+    ///@}
+
+    ///@name Protected Operators
+    ///@{
+    BushingElement() : Element()
+    {
+    }
+
+    ///@}
+
+private:
+    ///@name Private classes
+    ///@{
+
+    class Stiffness
+    {
+    public:
+        KRATOS_CLASS_POINTER_DEFINITION(Stiffness);
+
+        virtual ~Stiffness() = default;
+
+        virtual double GetValue(
+            const double Value,
+            const Properties& rProperties) const = 0;
+    };
+
+    class ConstantStiffness: public Stiffness
+    {
+    public:
+        KRATOS_CLASS_POINTER_DEFINITION(ConstantStiffness);
+
+        ConstantStiffness(const Variable<double>& rVariable) : mpVariable(&rVariable) {};
+
+        ~ConstantStiffness() = default;
+
+        double GetValue(
+            const double Value,
+            const Properties& rProperties) const override;
+
+    private:
+        Variable<double> const * mpVariable;
+    };
+
+    class NonLinearStiffness: public Stiffness
+    {
+    public:
+        KRATOS_CLASS_POINTER_DEFINITION(NonLinearStiffness);
+
+        NonLinearStiffness(
+            const Variable<double>& rVariableX,
+            const Variable<double>& rVariableY)
+            : mpVariableX(&rVariableX),
+              mpVariableY(&rVariableY)
+        {};
+
+        ~NonLinearStiffness() = default;
+
+        double GetValue(
+            const double Value,
+            const Properties& rProperties) const override;
+
+    private:
+        Variable<double> const * mpVariableX;
+
+        Variable<double> const * mpVariableY;
+    };
+
+    ///@}
+    ///@name Private member variables
+    ///@{
+
+    array_1d<Stiffness::UniquePointer, 6> mStiffnessGetters;
```

Comment:
why not std array?

Reply:
no specific reason... I can change it :)